src/backend/marsha/core/api.py

```diff
@@ -728,8 +728,12 @@ def get_queryset(self):
             if user.token.payload.get("user"):
                 if user.token.payload["user"].get("email"):
                     filters["email"] = user.token.payload["user"]["email"]
-                    return LiveRegistration.objects.filter(**filters)
-                # token has no email, user has access to this registration if it's the right
+                    # check first if a liveRegistration exists with the email in the token
+                    if LiveRegistration.objects.filter(**filters).count() > 0:
+                        return LiveRegistration.objects.filter(**filters)
```

Comment:
According to the [django documentation](https://docs.djangoproject.com/en/3.2/topics/db/optimization/#use-queryset-count), this code can be optimized using the following code

```suggestion
                    live_registration = LiveRegistration.objects.filter(**filters)
                    if live_registration:
                        return live_registration
```